gossip/sync.go

```diff
@@ -180,23 +185,32 @@ func (h *handler) snapsyncStageTick() {
 			if time.Since(bs.LastBlock.Time.Time()) > snapsyncMinEndAge {
 				continue
 			}
+			// cancel snapsync activity to prevent race condition
+			done := make(chan struct{})
+			h.snapState.updatesCh <- snapsyncStateUpd{
+				snapsyncCancelCmd: &snapsyncCancelCmd{done},
+			}
+			<-done
+			// finalize snapsync
 			err := h.process.SwitchEpochTo(epoch)
```

Comment:
I prefer this code 
```suggestion
			if err := h.process.SwitchEpochTo(epoch); err != nil {
                h.Log.Error("Failed to result snapsync", "epoch", epoch, "block", bs.LastBlock.Idx, "err", err)			
			}
			h.Log.Info("snapsync is finalized at", "epoch", epoch, "block", bs.LastBlock.Idx, "root", bs.FinalizedStateRoot)
```

this info log message `Epoch is switched by snapsync` sound a bit mis-leading like snapsync is switched to sync at new state root. Can we update to `snapsync is finalized at` or smth like that?